polkadot/xcm/xcm-executor/src/assets.rs

```diff
@@ -15,11 +15,12 @@
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
 use alloc::{
+	boxed::Box,
 	collections::{btree_map::BTreeMap, btree_set::BTreeSet},
 	vec::Vec,
 };
-use core::mem;
-use sp_runtime::{traits::Saturating, RuntimeDebug};
+use core::{fmt::Formatter, mem};
+use frame_support::traits::tokens::imbalance::ImbalanceAccounting;
```

Comment:
we break the rule of xcm-executor not depending on frame. to not do it we could have an adapter in the builder. not blocker for me, just heads-up.